elastalert/ircalert.py

```diff
@@ -0,0 +1,112 @@
+        if irc.client.is_channel(channel):
+            connection.join(channel)
+            self.logger.debug("Joined channel- connection.join(channel)")
+
+    def on_join(self, connection, event):
+        channel = self.channel
+        # Cannot privmsg newlines, nor longer than some maximum
+        message = self.body.replace("\n", " / ")[:200]
+        self.logger.debug("Recognized self.body, renamed as message - in on_join")
+        self.logger.debug("Recognized irc client object - in on_join")
+        if irc.client.is_channel(channel):
+            connection.privmsg(channel, message)
+            self.delivered = True
+            print("Alert message sent.")
+            self.logger.info("Message %s passed to IRC channel" % message)
+        else:
+            connection.reconnect()
+            self.logger.debug("Connection.reconnect()")
+            self.alert()
+            self.logger.info("Reconnected; tried to send message again.")
+
+    def on_privmsg(self, connection, event):
+        channel = self.channel
+        self.logger.debug("Recognized irc client object - in on_privmsg")
+        if irc.client.is_channel(channel):
+            connection.quit(message="I'm out, good luck!")
+            self.logger.debug("Disconnected from channel- connection.disconnect_all(message=..)")
+            connection.close()
+
+    def alert(self, matches):
+        client = self.client
+        server = client.server()
+        self.delivered = False
+        self.logger.debug("Recognized body of alert")
+        for match in matches:
+            self.body += " M:" + str(BasicMatchString(self.rule, match))
+            self.logger.debug("Appending matches from rule config to body of alert")
+            if len(matches) > 1:
+                self.body += "\n"
+                self.logger.debug("More to append to message body")
+        self.logger.debug("Body: '%s'" % self.body)
+        try:
+            c = server.connect(
+                self.server,
+                self.port,
+                self.botnick,
+                self.password,
+                self.username,
+                connect_factory=self.ssl_factory
+            )
+            self.logger.debug("Adding handlers")
+            c.add_global_handler("welcome", self.on_connect)
+            self.logger.debug("Added on_connect handler")
+            c.add_global_handler("join", self.on_join)
+            self.logger.debug("Added on_join handler")
+            c.add_global_handler("privmsg", self.on_privmsg)
+            self.logger.debug("Added on_privmsg handler")
+            status = c.is_connected()
+            if status is True:
+                self.logger.info("Connected to IRC: %s" % status)
+                while not self.delivered:
+                    client.process_once(0.2)
+                client.disconnect_all("done")
+                client.process_once(0.2)
+                self.logger.info("Done")
+                time.sleep(1)
```

Comment:
Why sleep at the end? Is this to prevent spamming the channel when there are multiple alerts?